src/components/search/Search.jsx

```diff
@@ -91,12 +98,57 @@ const Search = () => {
     [config.ALGOLIA_APP_ID, config.ALGOLIA_INDEX_NAME, config.ALGOLIA_SEARCH_API_KEY],
   );
 
+  // If a pathwayUUID exists, open the pathway modal.
+  useEffect(() => {
+    if (pathwayUUID) {
+      openLearnerPathwayModal();
+    }
+  }, [openLearnerPathwayModal, pathwayUUID]);
+
   const PAGE_TITLE = `${HEADER_TITLE} - ${enterpriseConfig.name}`;
+
+  // Determine whether learner has only content assignments available to them, based on the presence of:
+  // - active content assignments
+  // - no auto-applied budgets
+  // - no current enterprise offers
+  // - no active license or license requests
+  // - no assigned codes or code requests
+  const hasActiveLicense = subscriptionPlan && subscriptionLicense?.status === LICENSE_STATUS.ACTIVATED;
+  const hasActiveLicenseOrLicenseRequest = hasActiveLicense || licenseRequests.length > 0;
+  const hasAssignedCodesOrCodeRequests = couponCodesCount > 0 || couponCodeRequests.length > 0;
+  const hasAutoAppliedLearnerCreditPolicies = redeemableLearnerCreditPolicies?.filter(
+    policy => policy.policyType !== POLICY_TYPES.ASSIGNED_CREDIT,
+  ).length > 0;
+  const hasContentAssignmentsOnly = !!(
+    learnerContentAssignments.hasActiveAssignments
+    && !hasCurrentEnterpriseOffers
+    && !hasActiveLicenseOrLicenseRequest
+    && !hasAssignedCodesOrCodeRequests
+    && !hasAutoAppliedLearnerCreditPolicies
+  );
+
+  useEffect(() => {
+    if (hasContentAssignmentsOnly) {
+      hideRecommendCourses();
+    } else {
+      showRecommendCourses();
+    }
+  }, [showRecommendCourses, hideRecommendCourses, hasContentAssignmentsOnly]);
```

Comment:
This `useEffect` is responsible for the hiding/showing of the "Recommend courses for me" (Skills Quiz) CTA in the `EnterpriseBanner`.